JASP-Engine/JASP/R/correlation.R

```diff
@@ -15,806 +15,721 @@
-											.displayError(errorMessagePlot, cexText=cexText)
-										}
-
-									} else {
-
-										plot(1, type= "n", axes= FALSE, ylab="", xlab="")
+					estimate <- as.numeric(result$estimate)
+					p.value  <- as.numeric(result$p.value)
+					MPR <- .VovkSellkeMPR(p.value)
 
-									}
-								}
+					if (test == "pearson") {
 
-								if (l >= 7) {
+						upperCI <- as.numeric(result$conf.int[2])
+						lowerCI <- as.numeric(result$conf.int[1])
 
-									if (options$plotStatistics) {
+					} else if (test == "spearman") {
 
-										if ( ! variable.statuses[[col]]$unplotable && ! variable.statuses[[row]]$unplotable) {
-											.plotCorValue(dataset[[variables[col]]], dataset[[variables[row]]], cexCI= 1.2, hypothesis= options$hypothesis,
-											pearson=options$pearson, kendallsTauB=options$kendallsTauB, spearman=options$spearman, confidenceInterval=options$confidenceIntervalsInterval)
-										} else {
-											errorMessages <- c(variable.statuses[[row]]$plottingError, variable.statuses[[col]]$plottingError)
-											errorMessagePlot <- paste0("Correlation coefficient undefined:", "\n", errorMessages[1])
-											.displayError(errorMessagePlot, cexText=cexText)
-										}
+					  spearCI <- .createNonparametricConfidenceIntervals(obs1, obs2, method = "spearman", hypothesis = hypothesis, confLevel = CI)
+					  upperCI <- as.numeric(spearCI[2])
+					  lowerCI <- as.numeric(spearCI[1])
 
-									} else {
+					} else if (test == "kendall") {
 
-										plot(1, type= "n", axes= FALSE, ylab="", xlab="")
+					  kendallCI <- .createNonparametricConfidenceIntervals(obs1, obs2, method = "kendall", hypothesis = hypothesis, confLevel = CI)
+					  upperCI <- as.numeric(kendallCI[2])
+					  lowerCI <- as.numeric(kendallCI[1])
 
-									}
-								}
-							}
-						}
 					}
-				}
-
-
-				if (l > 2 || ((l == 2 && options$plotDensities) || (l == 2 && options$plotStatistics))) {
-
-					textpos <- seq(1/(l*2), (l*2-1)/(l*2), 2/(l*2))
-
-					if (!options$plotDensities && !options$plotStatistics) {
-
-							for (t in seq_along(textpos)) {
-
-								mtext(text = .unv(variables)[t], side = 3, outer = TRUE, at= textpos[t], cex=1.5, line= -0.8)
-
-								if (t < length(textpos)) {
 
-									mtext(text = .unv(variables)[t], side = 2, outer = TRUE, at= rev(textpos)[t], cex=1.5, line= -0.1, las= 0)
-
-								}
-							}
-
-					} else {
-
-						for (t in seq_along(textpos)) {
-
-								mtext(text = .unv(variables)[t], side = 3, outer = TRUE, at= textpos[t], cex=1.5, line= -0.8)
-								mtext(text = .unv(variables)[t], side = 2, outer = TRUE, at= rev(textpos)[t], cex=1.5, line= -0.1, las= 0)
-						}
-					}
-				}
-				}
+        }
 
-				# content <- .endSaveImage(image)
-				content <- .writeImage(width = width, height = height, plot = .plotFunc, obj = TRUE)
+      }
 
-				plot <- correlation.plot
+      results[["result"]][[pairName]][[test]] <- list(
```

Comment:
saving the cleaned result might clash with significant numbers later on, which is why I would save the raw object here. Or do you think it's better to clean them instead?

Reply:
.clean doesn't change numeric values